File: NaoBlocks.Web/Dtos/SystemStatus.cs
using System.Collections.Generic;

namespace NaoBlocks.Web.Dtos
{
    public class SystemStatus
    {
        public IList<RobotStatus> RobotsConnected { get; } = new List<RobotStatus>();

        public IList<UserStatus> UsersConnected { get; } = new List<UserStatus>();
    }
}

File: client/src/app/data/system-status.ts
import { UserStatus } from './user-status';
import { RobotStatus } from './robot-status';

export class SystemStatus {
    usersConnected: UserStatus[] = [];
    robotsConnected: RobotStatus[] = [];
    error: string;

    constructor(error?: string) {
        this.error = error;
    }
}
